Cosmetic changes and cleaning of default params.

ruriky opened the pull request:
- Removed default values for optional parameters / changed default values to match openldaps own default values. We should let the openldap handle the default values for optional params.
- Removed one un-used parameter "$schema_dir".

Puppet-lint founds:
- Added small header comment for all classes / defines.
- Double quotes -> single quotes.
- Removed trailing spaces.

ruriky committed:
Cosmetic changes and cleaning of default params.

- Removed default values of optional parameters (params.pp). We should use the
  values which come from the openldap slapd itself.
- Modified templates to ignore parameters which are not defined.
- Removed one un-used parameter.

Puppet-lint founds:
- Added small header comments for all classes / defines.
- Double quotes -> single quotes.
- Removed trailing spaces.

crayfishx commented on manifests/init.pp:
```diff
@@ -47,31 +47,30 @@
   $modules                 = [],
-  $password_hash           = $openldap_slapd::params::password_hash,
-  $password_salt_format    = $openldap_slapd::params::password_salt_format,
+  $password_hash           = undef,
```
We cannot undef some of these defaults, as some of them aren't surrounded by conditionals in the template...  eg `password_hash`..

https://github.com/crayfishx/puppet-openldap_slapd/blob/master/templates/_security.erb#L4

This will cause the template to spit out just 

```
password-hash
```

or 

```
password-hash :undef
```

Depending on which version of Puppet is being used.   

The manifest changes are fine but we need to check how each one is handled in the template.  Keep in mind the behaviour differences between puppet 3 and 4 when it comes to handling "undef" variables in ERB templates....

ruriky commented:
Hi,

Thanks for noticing the mistake! I was a bit too hasty pushing my changes.

I have now tried to surround all the optional variables with if conditionals, so that parameters with "undef" values won't be placed into the config file. I could find only two of those: $password_hash and $security.

I also added a changes to improve Puppet 4 compatibility:

``` Diff
-  if $tls_enabled == true {
+  if str2bool("$tls_enabled") {
```

.. and removed some commented code I had left behind and unified the header comments I had added before.
